packages/harpguru-core/src/components/harp-guru/hooks/use-flush-buffered-activity-toggles/use-flush-buffered-activity-toggles.ts

```diff
@@ -0,0 +1,43 @@
+import { useGlobal } from 'reactn'
+import { unstable_batchedUpdates } from 'react-native'
+import { useEffect } from 'react'
+import type { HarpStrataProps } from 'harpstrata'
+import { getHarpStrata } from 'harpstrata'
+
+import { batchToggleActiveIds } from '../../utils'
+
+export const useFlushBufferedActivityToggles = (): void => {
+  const [bufferedActivityToggles, setBufferedActivityToggles] = useGlobal(
+    'bufferedActivityToggles'
+  )
+  const [activeHarpStrata, setActiveHarpStrata] = useGlobal('activeHarpStrata')
+
+  useEffect(() => {
+    const flushBufferedToggles = setTimeout(() => {
+      if (bufferedActivityToggles.length === 0) return
+      const {
+        apparatus: { id: apparatusId },
+        pozitionId,
+        harpKeyId,
+        isActiveComplex: { activeDegreeIds },
+      } = activeHarpStrata
+      const newActiveDegreeIds = batchToggleActiveIds(
+        activeDegreeIds,
+        bufferedActivityToggles
+      )
+      const newHarpStrataProps: HarpStrataProps = {
+        apparatusId,
+        pozitionId,
+        harpKeyId,
+        activeIds: newActiveDegreeIds,
+      }
+      unstable_batchedUpdates(() => {
+        setActiveHarpStrata(getHarpStrata(newHarpStrataProps))
```

Comment:
Missing some of these functions in the deps. Need to consider whether they can be excluded.

Reply:
Updated, but I don't think it's necessary and needs to be recorded as an opportunity to optimise.